File: sdk/swift/Sources/SeocheonSDK/Types/NodeStatus.swift
import Foundation

/// Represents the status of a registered node.
public enum NodeStatus: String, Codable, Sendable {
    case unspecified = "UNSPECIFIED"
    case registered = "REGISTERED"
    case active = "ACTIVE"
    case inactive = "INACTIVE"
    case jailed = "JAILED"

    /// Converts a proto enum integer to NodeStatus.
    public static func fromInt(_ value: Int) -> NodeStatus {
        switch value {
        case 1: return .registered
        case 2: return .active
        case 3: return .inactive
        case 4: return .jailed
        default: return .unspecified
        }
    }

    /// Converts a protobuf JSON status string to NodeStatus.
    /// Accepts both "NODE_STATUS_REGISTERED" and "REGISTERED" formats.
    public static func fromString(_ value: String) -> NodeStatus {
        let s = value.hasPrefix("NODE_STATUS_")
            ? String(value.dropFirst("NODE_STATUS_".count))
            : value
        switch s {
        case "REGISTERED": return .registered
        case "ACTIVE":     return .active
        case "INACTIVE":   return .inactive
        case "JAILED":     return .jailed
        default:           return .unspecified
        }
    }
}

File: sdk/kotlin/src/main/kotlin/com/seocheon/sdk/types/NodeStatus.kt
package com.seocheon.sdk.types

/**
 * Represents the status of a registered node.
 */
enum class NodeStatus(val value: String) {
    UNSPECIFIED("UNSPECIFIED"),
    REGISTERED("REGISTERED"),
    ACTIVE("ACTIVE"),
    INACTIVE("INACTIVE"),
    JAILED("JAILED");

    companion object {
        /**
         * Converts a proto enum integer to NodeStatus.
         */
        fun fromInt(status: Int): NodeStatus = when (status) {
            1 -> REGISTERED
            2 -> ACTIVE
            3 -> INACTIVE
            4 -> JAILED
            else -> UNSPECIFIED
        }

        /**
         * Converts a string to NodeStatus, case-insensitive.
         */
        fun fromString(status: String): NodeStatus =
            entries.find { it.value.equals(status, ignoreCase = true) } ?: UNSPECIFIED
    }
}
